test/integration/nighthawk_test_server.py

```diff
@@ -27,45 +30,63 @@ def __init__(self, server_binary_path, config_template_path, server_ip, server_p
     self.server_process = None
     self.server_ip = server_ip
     self.socket_type = socket.AF_INET6 if ip_version == IpVersion.IPV6 else socket.AF_INET
-    self.server_port = server_port
+    self.server_port = -1
+    self.admin_port = -1
+    self.admin_address_path = ""
+    self.parameterized_config_path = ""
+    self.instance_id = str(random.randint(1, 1024 * 1024 * 1024))
     self.parameters = parameters
     self.server_binary_config_path_arg = server_binary_config_path_arg
 
-  def serverThreadRunner(self):
     self.parameters["server_ip"] = self.server_ip
-    self.parameters["server_port"] = self.server_port
-
     with open(self.config_template_path) as f:
       config = Template(f.read())
       config = config.substitute(self.parameters)
-
-    parameterized_config_path = None
     with tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".yaml") as tmp:
-      parameterized_config_path = tmp.name
+      self.parameterized_config_path = tmp.name
       tmp.write(config)
+    with tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".adminpath") as tmp:
+      self.admin_address_path = tmp.name
 
+  def serverThreadRunner(self):
     args = [
-        self.server_binary_path, self.server_binary_config_path_arg, parameterized_config_path,
-        "-l", "error", "--base-id",
-        str(self.server_port)
+        self.server_binary_path, self.server_binary_config_path_arg, self.parameterized_config_path,
+        "-l", "error", "--base-id", self.instance_id, "--admin-address-path",
+        self.admin_address_path
     ]
     logging.info("Test server popen() args: [%s]" % args)
     self.server_process = subprocess.Popen(args)
     self.server_process.communicate()
 
-  def waitUntilServerListening(self):
-    with socket.socket(self.socket_type, socket.SOCK_STREAM) as sock:
-      sock.settimeout(1)
-      tries = 10
-      while tries > 0:
-        time.sleep(0.5)
-        tries -= 1
-        if sock.connect_ex((self.server_ip, self.server_port)) == 0:
-          return True
-      logging.error("Timeout while waiting for server listener at %s:%s to accept connections.",
-                    self.server_ip, self.server_port)
+  def tryUpdateFromAdminInterface(self):
```

Comment:
This is great.. my only question is would it make sense to have a common admin access method that can do both stats scraping and listener fetching etc?

Reply:
Yes; Done in 5109fd408c2bbf62fd2e188c12c7c04bf4030fe2